module1/homework/grayscaleImages/compression.hpp

```diff
@@ -0,0 +1,11 @@
+#include <vector>
+#include <array>
+
+namespace{
+    constexpr uint8_t width = 32;
+    constexpr uint8_t height = 32;
+}
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(std::array<std::array<uint8_t, width>, height>&);
+std::array<std::array<uint8_t, width>, height> decompressGrayscale(std::vector<std::pair<uint8_t, uint8_t>>&);
+void printMap(std::array<std::array<uint8_t, width>, height>&);
```

Comment:
Have you considered using alias for longer  types like std::vector<std::pair<uint8_t, uint8_t>>?